### ISSUE TITLE
Can't know if preview in transform data userFunc

### ISSUE THREAD

**Issue by webian** [cobwebch/external_import#105]
TLDR:
It's missing the possibility to know if in preview mode inside transform data userFunc. Maybe pass a boolean to userFunc with the state of the preview (on/off)?

Long story:
I'm using this great extension for a special case: external system (ERP) send xml with data of products to TYPO3 that will import it.
The products in the xml have categories but these have not an ID, only the title. So I can't import the categories in a specific import, I have to create them while importing the products. 
To do it I use transform data userFunc to create the categories with the title specified in the category field of the product.
So far so good, but I would like to avoid to create the categories if in preview mode and, if I'm not wrong, it's not possible because there's no way to know if the userFunc is run in preview mode or not.

**Comment by fsuter** [comment]
I can see how that would be useful indeed. More generally, I think it would be useful that user functions have a reference to the `Importer` instance, since that allows so many things. I'll have to check how best to implement that without breaking existing user functions.

**Comment by fsuter** [comment]
BTW which TYPO3 version are you running. My current intent is that the next version of external_import will support only v9.

**Comment by webian** [comment]
Right! Would be even better to have the `Importer` instance available.

I'm still on TYPO3 8.7 atm.

**Comment by fsuter** [comment]
Done using an interface and a trait, and that's fully backwards-compatible, since only user functions implementing the interface get the `Importer` instance. And will be available in a TYPO3 8-compatible release :-)
